### ISSUE TITLE
Setting CollectionView.SelectedItem = null does nothing

### ISSUE THREAD

**Issue by mattleibow** [dotnet/maui#10025]
### Description

I have a simple collection view that is bound to a VM and I am trying to set (unset actually) the SelectedItem to null.

I tried setting the bound property on the VM and the actual `vc.SlectedItem = null` but it remains selected. 

I tried setting the SelectionMode to None, and this causes the property to now return null, but setting it back to Single causes it to "reselect".

### Steps to Reproduce

TODO

### Link to public reproduction project repository

TODO

### Version with bug

6.0.486 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Windows, I was *not* able test on other platforms

### Affected platform versions

All

**Comment by XamerDev** [comment]
is there any workaround for this problem? 
I use collview for few items and it is big problem because user have to select different item and then go to the first one to choose it

**Comment by IsmailHassani** [comment]
Try setting it to null first. That worked for me. But it is bs.

**Comment by dotMorten** [comment]
I just hit this exact issue as well and is blocking me from using collectionview

**Comment by Foda** [comment]
Hmm... I'm having a hard time reproducing this on `main`. @dotMorten do you have a repro I could take a look at?

I've got a `CollectionView` bound to a list of strings, w/ bound `SelectedItem` too. Then a button clears the selected item in the VM.

**Comment by dotMorten** [comment]
@Foda You're using Windows and .NET 7 right?

**Comment by Foda** [comment]
@dotMorten Yup -- wondering if this was fixed but not yet released. Lemme double check on the release branch.

**Comment by dotMorten** [comment]
I'll see if I can't get a simple repro I can share (the one I have is pretty involved, and the public branch no longer uses the selected item because of this bug. Anyway, here's a recording of hitting the SelectionChanged event, checking that the selected item has a value, then setting it to null and checking that it isn't null but still has the old value:
https://github.com/dotnet/maui/assets/1378165/938bc7c1-ef93-4b41-9811-1d1a259e706f

**Comment by dotMorten** [comment]
Simple sample: File new maui app. Replace `MainPage` content with:
`<CollectionView x:Name="cv" SelectionMode="Single" />`
In code-behind replace with:
```cs
        public MainPage()
        {
            InitializeComponent();
            cv.ItemsSource = new string[] { "Line1", "Line2", "Line3", "Line4", "Line5" };
            cv.SelectionChanged += Cv_SelectionChanged;
        }

        private void Cv_SelectionChanged(object sender, SelectionChangedEventArgs e)
        {
            cv.SelectedItem = null;
        }
```
Now if you step through the selection event handler, you'll notice SelectedItem isn't null, but the event handler does get hit a second time now stating it is null. So it's like the property gets a delayed update. However if you look in the view, the item still appears selected, and you can't click it again - you have to click something else.

**Comment by dotMorten** [comment]
Oh it's suddenly clear why it isn't working. It was never implemented: https://github.com/dotnet/maui/blob/098a9fa95230414cf782eb0f3ac90c23a15d1239/src/Controls/src/Core/Handlers/Items/SelectableItemsViewHandler.Windows.cs#L64-L74

**Comment by Foda** [comment]
Okay, I see what's happening here. It's implemented, but in the function `UpdatePlatformSelection`. WPF does have the same odd behavior, but the event will fire immediately when the selection changes to `null` rather than wait

**Comment by Zhanglirong-Winnie** [comment]
Verified this issue with Visual Studio Enterprise 17.9.0 Preview 2. Can repro this issue.

**Comment by the-netspider** [comment]
I CAN CONFIRM that **'Collectionview.SelectedItem = null'** is still not working on Windows. 
But Its working fine on Android

Once the Item is Selected You cannot Unselect or make it null with any Method. You have to select another item to unselect it.. and its very Frustrating.

**Comment by assurasc** [comment]
I can confirm that this is still not working on Windows. I had to change over to a ListView that works correctly. Any news on if and when this will get fixed?

**Comment by assurasc** [comment]
Here is a repro for the issue [https://github.com/stevechadbourne/CollectionViewIssue](url)
Put a breakpoint in the MainPageModel.SelectedNumber setter.
Running on Windows - tap a number in the list - breakpoint is hit. However list item is not unhighlighted and tapping it again does not hit breakpoint.
Running on Android - list item is unselected and tapping it a second time hits breakpoint.

**Comment by assurasc** [comment]
> Here is a repro for the issue [https://github.com/stevechadbourne/CollectionViewIssue](url)
> Put a breakpoint in the MainPageModel.SelectedNumber setter.
> Running on Windows - tap a number in the list - breakpoint is hit. However list item is not unselected and tapping it again does not hit breakpoint.
> Running on Android - list item is unselected and tapping it a second time hits breakpoint.



> > I can confirm that this is still not working on Windows. I had to change over to a ListView that works correctly. Any news on if and when this will get fixed?
> 
> Can you please provide a repro? My very simple "list with items, button to clear" seems to work fine:
> 
>  Recording.2024-06-21.131216.mp4

Here is a repro for the issue [https://github.com/stevechadbourne/CollectionViewIssue](https://github.com/dotnet/maui/issues/url)
Put a breakpoint in the MainPageModel.SelectedNumber setter.
Running on Windows - tap a number in the list - breakpoint is hit. However list item is not unselected and tapping it again does not hit breakpoint.
Running on Android - list item is unselected and tapping it a second time hits breakpoint.

**Comment by kitecraft** [comment]
My workaround is to reset the list when on windows, which causes the selectedItem to be reset.  It causes a visual glitch, but at least it works.
...
                SelectedDevice = null;
#if WINDOWS
                DeviceList = new List<WifiScan_Device>(DeviceList);
#endif

**Comment by Saurbaum** [comment]
Really annoying that this is still present

**Comment by Marcos-Jose-DV** [comment]
It appears to use a delay to select and deselect, thus being able to select the same item from the list.

```Csharp
  await Task.Delay(100);
  folderCollectionView.SelectedItem = null;
```

**Comment by assurasc** [comment]
7 months since it was under consideration. Please can you bump this up the priority list @PureWeen as its still causing issues for people.
